backend-api/src/functions/adapters/aws/dynamo/dynamoDbAdapter.test.ts

```diff
@@ -293,4 +304,129 @@ describe("DynamoDbAdapter", () => {
+        result = await sessionRegistry.getSession(getOperation);
+      });
+
+      it("Logs the failure", () => {
+        expect(consoleErrorSpy).toHaveBeenCalledWithLogFields({
+          messageCode: "MOBILE_ASYNC_GET_SESSION_UNEXPECTED_FAILURE",
+        });
+      });
+
+      it("Returns failure with server error", () => {
+        expect(result).toEqual(
+          errorResult({
+            errorType: GetSessionError.INTERNAL_SERVER_ERROR,
+          }),
+        );
+      });
+    });
+
+    describe("Given session was not found", () => {
+      beforeEach(async () => {
+        mockDynamoDbClient.on(GetItemCommand).resolves({});
+        result = await sessionRegistry.getSession(getOperation);
+      });
+
+      it("Logs the failure", () => {
+        expect(consoleErrorSpy).toHaveBeenCalledWithLogFields({
+          messageCode: "MOBILE_ASYNC_GET_SESSION_SESSION_NOT_FOUND",
+        });
+      });
+
+      it("Returns failure with an invalid session error", () => {
+        expect(result).toEqual(
+          errorResult({
+            errorType: GetSessionError.SESSION_NOT_FOUND,
+          }),
+        );
+      });
+    });
+
+    describe("Given session was found", () => {
+      describe("Given invalid session attributes were returned in response", () => {
+        beforeEach(async () => {
+          mockDynamoDbClient.on(GetItemCommand).resolves({
+            Item: marshall({
+              clientId: "mockClientId",
+              govukSigninJourneyId: "mockGovukSigninJourneyId",
+              createdAt: 12345,
+              issuer: "mockIssuer",
+              sessionId: mockSessionId,
+              sessionState: SessionState.AUTH_SESSION_CREATED,
+              clientState: "mockClientState",
+              subjectIdentifier: "mockSubjectIdentifier",
+              timeToLive: 12345,
+            }),
+          });
+          result = await sessionRegistry.getSession(getOperation);
+        });
+
+        it("Logs the failure", () => {
+          expect(consoleErrorSpy).toHaveBeenCalledWithLogFields({
+            messageCode: "MOBILE_ASYNC_GET_SESSION_SESSION_NOT_FOUND",
+          });
+        });
+
+        it("Returns failure with server error", () => {
+          expect(result).toEqual(
+            errorResult({
+              errorType: GetSessionError.SESSION_NOT_FOUND,
+            }),
+          );
+        });
+      });
+
+      describe("Given valid session attributes were returned in response", () => {
+        const validBiometricTokenIssuedSessionAttributesItem = marshall({
+          ...validBiometricTokenIssuedSessionAttributes,
+          createdAt: 1704106860000, // 2024-01-01T11:01:00.000Z
+        });
+
+        beforeEach(async () => {
+          mockDynamoDbClient.on(GetItemCommand).resolves({
+            Item: validBiometricTokenIssuedSessionAttributesItem,
+          });
+          result = await sessionRegistry.getSession(getOperation);
+        });
+
+        it("Logs the failure", () => {
```

Comment:
```suggestion
        it("Logs the success", () => {
```

or something similar